WooCommerce/Classes/Blaze/BlazeEligibilityChecker.swift

```diff
@@ -0,0 +1,75 @@
+import Foundation
+import Yosemite
+
+/// Protocol for checking Blaze eligibility for easier unit testing.
+protocol BlazeEligibilityCheckerProtocol {
+    func isEligible() async -> Bool
+    func isEligible(product: ProductFormDataModel, isPasswordProtected: Bool) async -> Bool
```

Comment:
Nit: It's not obvious from the naming that the first check is for sites. Should we mention that in the method names?

```suggestion
    func checkSiteEligibility() async -> Bool
    func checkProductEligibility(_ product: ProductFormDataModel, isPasswordProtected: Bool) async -> Bool
```

Reply:
I thought about renaming it mid way but thought I'd wait for the feedback 😝 In https://github.com/woocommerce/woocommerce-ios/pull/9899/commits/c4ed37bdbd50da7e9e31e6967a83330ecb4803fc, renamed it to `isSiteEligible` and `isProductEligible`.